.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+        exclude: '(\.pdb$|\.sdf$|\.proto$|\.xml$)'
+    -   id: trailing-whitespace
+        exclude: '(\.pdb$|\.sdf$|\.proto$|\.xml$)'
+
+-   repo: https://github.com/psf/black
+    rev: 21.10b0
+    hooks:
+    -   id: black
+        exclude: '^(ff/params/|parallel/grpc/)'
+        args: [--line-length=120]
+
+-   repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v13.0.0
+    hooks:
+    -   id: clang-format
+        include: '^timemachine/cpp/src/'
+        exclude: '^timemachine/cpp/src/jitify.hpp'
```

Comment:
We might also want to exclude the hilbert.c/.hpp files if the intention is to avoid modifying vendored code

Reply:
Good call; this is fixed now (rebased to keep formatting changes in one commit)